### PR TITLE
Fixed the zero filter and finalizeLocalRomSegmentEvaluation index

### PR DESCRIPTION
**Pull request by JiaZhou-PU**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)

When Long term Fourier assigned zero-filter might over detrending the data.
Picker of adding the Fourier back is wrong   
Preserve CDF of limited training data need to be scaled

##### What are the significant changes in functionality due to this change request?

Zero filter mask now means the data that are all zero.

Change the double mask of interpolateDist method in ARMA into a tuple mask to remove the 
future warning:
FutureWarning: Using a non-tuple sequence for multidimensional indexing is deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`

Add the evaluate scale in multiyear
----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [ ] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: framework/SupervisedLearning/ARMA.py [hunk 1]
```diff
@@ -502,6 +498,18 @@ def __evaluateLocal__(self, featureVals):
...
+  def _evaluateScale(self, growthInfo,year):
+    """
+    """
+    growth = growthInfo['value']/100. # given in percentage
+    mode = growthInfo['mode']
+    if mode == 'exponential':
+      scale = (1.0 + growth) ** year
+    else:
+      scale = 1.0 + year * growth
+    # print('jialock holmes is growing',scale)
```
Comment (PaulTalbot-INL) [inline framework/SupervisedLearning/ARMA.py]: this can be removed

File: framework/SupervisedLearning/ARMA.py [hunk 2]
```diff
@@ -1957,7 +1969,37 @@ def finalizeLocalRomSegmentEvaluation(self, settings, evaluation, picker):
       for target, signal in settings['long Fourier signal'].items():
         ## NOTE might need to put zero filter back into it
         sig = signal['predict'][picker]
-        evaluation[target][picker] += sig
+        # print('jialock holmes is inside finalize local rom segment',target,picker)
+        # print('jialock holmes is inside finalize local rom segment',len(evaluation['GHI'][0]))
+        # sigLen=picker.stop-picker.start
```
Comment (PaulTalbot-INL) [inline framework/SupervisedLearning/ARMA.py]: extra lines

File: framework/SupervisedLearning/ARMA.py [hunk 3]
```diff
@@ -1957,7 +1969,37 @@ def finalizeLocalRomSegmentEvaluation(self, settings, evaluation, picker):
...
+          if bgId is not None:
+            sigLen=picker.stop-picker.start
+            bgInd = bgId
+            endInd = bgInd+sigLen
+            for y in range(len(evaluation[target])):
+              evaluation[target][y][bgInd:endInd] += sig
+          else:
+            for y in range(len(evaluation[target])):
+              evaluation[target][y][picker] += sig
+        # pp.pprint(evaluation)
```
Comment (PaulTalbot-INL) [inline framework/SupervisedLearning/ARMA.py]: extra line

File: framework/SupervisedLearning/ARMA.py [hunk 4]
```diff
@@ -1957,7 +1969,37 @@ def finalizeLocalRomSegmentEvaluation(self, settings, evaluation, picker):
...
+        # if multiyear
+        else:
+          # print('lalalallalaal')
```
Comment (PaulTalbot-INL) [inline framework/SupervisedLearning/ARMA.py]: extra line

File: framework/SupervisedLearning/ARMA.py [hunk 5]
```diff
@@ -1957,7 +1969,37 @@ def finalizeLocalRomSegmentEvaluation(self, settings, evaluation, picker):
...
+
+          if bgId is not None:
+            # print(picker)
+            # print(len(evaluation[target]))
```
Comment (PaulTalbot-INL) [inline framework/SupervisedLearning/ARMA.py]: extra lines

File: framework/SupervisedLearning/ARMA.py [hunk 6]
```diff
@@ -1957,7 +1969,37 @@ def finalizeLocalRomSegmentEvaluation(self, settings, evaluation, picker):
...
+            sigLen=picker.stop-picker.start
+            bgInd = bgId
+            endInd = bgInd+sigLen
+            # print(sigLen)
+            # print(bgInd,endInd)
```
Comment (PaulTalbot-INL) [inline framework/SupervisedLearning/ARMA.py]: extra lines

File: framework/SupervisedLearning/ARMA.py [hunk 7]
```diff
@@ -1972,8 +2014,17 @@ def finalizeGlobalRomSegmentEvaluation(self, settings, evaluation, weights=None)
     # backtransform signal
     ## how nicely does this play with zerofiltering?
     if self.preserveInputCDF:
+      # print('jialock holmes is detecting pcdf',settings['input CDFs'])
```
Comment (PaulTalbot-INL) [inline framework/SupervisedLearning/ARMA.py]: extra line

**Comment by JiaZhou-PU** [comment]
@PaulTalbot-INL ready for review.

**Comment by PaulTalbot-INL** [comment]
It looks like the file `raven/tests/framework/ROM/TimeSeries/ARMA/Peaks/hrl.csv` is missing for the `ARMA/Peaks` test.

**Review by PaulTalbot-INL — COMMENTED** [review]
Code changes are fine to here, but manually running the ARMA tests I see missing files for the Peak test.

**Comment by JiaZhou-PU** [comment]
> Code changes are fine to here, but manually running the ARMA tests I see missing files for the Peak test.

Sorry, checkout the original test data file now. @PaulTalbot-INL

**Comment by PaulTalbot-INL** [comment]
Some tests are failing, but likely not due to the particular changes you made.

Merging into development branch.

**Pull request merged**